modules/core/src/main/java/org/apache/fluo/core/impl/TransactionImpl.java

```diff
@@ -203,12 +203,17 @@ public Bytes get(Bytes row, Column column) {
     ParallelSnapshotScanner pss = new ParallelSnapshotScanner(rowColumns, env, startTs, stats);
 
     Map<Bytes, Map<Column, Bytes>> ret = pss.scan();
+    Map<RowColumn, Bytes> bet = new HashMap<>();
 
     for (Entry<Bytes, Map<Column, Bytes>> entry : ret.entrySet()) {
       updateColumnsRead(entry.getKey(), entry.getValue().keySet());
+      for (Entry<Column, Bytes> cols : entry.getValue().entrySet()) {
+        // create RowColumn and add to new map
+        bet.put(new RowColumn(cols.getValue()), cols.getValue());
```

Comment:
`cols` should probably be called `colVal`.  You need the row and col.  The entry in the outer loop has the row.  The entry in the inner loop has col and value.

```java

     for (Entry<Bytes, Map<Column, Bytes>> entry : ret.entrySet()) {
       updateColumnsRead(entry.getKey(), entry.getValue().keySet());
       for (Entry<Column, Bytes> colVal : entry.getValue().entrySet()) {
         Bytes row = entry.getKey();
         Column col = colVal.getKey();
         Bytes val = colVal.getValue();
         bet.put(new RowColumn(row, col), val);
         

```